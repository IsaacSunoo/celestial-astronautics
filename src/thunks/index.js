import { setLoading, getApolloData, setError, getDaily } from '../actions';
import { fetchCall } from '../api';
import apiKey from '../api_key/apiKey';

// const baseUrl = 'https://api.nasa.gov/';

export const getApollo20 = () => {
    const url = 'https://images-api.nasa.gov/search?q=apollo%20landing&media_type=image';
    return async dispatch => {
        try {
            dispatch(setLoading(true));
            const apollo = await fetchCall(url);
            dispatch(setLoading(false));
            dispatch(getApolloData(apollo));
        } catch (err) {
            dispatch(setError(err.message));
        }
    }
}

export const getDailyNews = () => {
    const url = `https://api.nasa.gov/planetary/apod?api_key=${apiKey}`;
    return async dispatch => {
        try {
            dispatch(setLoading(true));
            const dailyNews = await fetchCall(url);
            dispatch(setLoading(false));
            dispatch(getDaily(dailyNews));
        } catch (err) {
            dispatch(setError(err.message));
        }
    }
}

