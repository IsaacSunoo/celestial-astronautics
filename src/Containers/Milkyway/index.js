import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getMilkyway } from '../../thunks';
import LoadingGif from '../../images/Loading.gif';
import InfoCard from '../../components/InfoCard';
import Particles from '../../components/Particles';

class Milkyway extends Component {

    componentDidMount = async () => {
        const { getMilkyway } = this.props;
        await getMilkyway();
    }

    render() {
        const { milkyway, isLoading } = this.props;
        let displayMilkyway;
        if (milkyway.length !== 0) {
            displayMilkyway = milkyway.items.map(milkyItem => {
                return (
                    <div key={milkyItem.data[0].nasa_id}>
                        <InfoCard information={milkyItem} />
                    </div>
                )
            });
        }

        return (
            isLoading ?
            (
            <div className='loading-display'>
                <h2>Loading</h2>
                <img src={LoadingGif} alt='loading gif' />
            </div>
            ) :
            (
            <div className='info-items-container'>
                <Particles />
                {displayMilkyway}
            </div>
            )
        )
    }
}

export const mapStateToProps = state => ({
    milkyway: state.milkyway,
    isLoading: state.isLoading
});

export const mapDispatchToProps = dispatch => ({
    getMilkyway: () => dispatch(getMilkyway())
});


export default connect(mapStateToProps, mapDispatchToProps)(Milkyway);