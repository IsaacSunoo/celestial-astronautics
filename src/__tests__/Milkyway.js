import React from 'react';
import { Milkyway, mapStateToProps, mapDispatchToProps } from '../containers/Milkyway';
import { shallow } from 'enzyme';
import { getMilkyway } from '../thunks';

jest.mock('../thunks');
const mockMilkyway = [];
const mockIsLoading = false;

describe('milkyway', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Milkyway milkyway={mockMilkyway} isLoading={mockIsLoading} />);
  });

  it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('returns an object with houses', () => {
      const mockState = {
        milkyway: ['constellation', 'black hole'],
        isLoading: false,
        error: 'error'
      };
      const expectedState = {
        milkyway: ['constellation', 'black hole'],
        isLoading: false
      };
      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedState);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a getMilkyway thunk', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = getMilkyway();
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.getMilkyway();
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
