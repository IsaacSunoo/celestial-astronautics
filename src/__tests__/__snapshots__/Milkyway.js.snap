// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`milkyway should match snapshot 1`] = `
<div
  className="info-items-container"
>
  <ParticlesContainer />
</div>
`;
